grouper/ctl/sync_db.py

```diff
@@ -9,88 +9,94 @@
-    # Models not implicitly or explictly imported above are explicitly imported here
-    from grouper.models.perf_profile import PerfProfile  # noqa: F401
-    from grouper.models.user_token import UserToken  # noqa: F401
-
-    db_engine = get_db_engine(settings.database_url)
-    Model.metadata.create_all(db_engine)
-
-    # Add some basic database structures we know we will need if they don't exist.
-    session = make_session(settings)
-
-    for name, description in SYSTEM_PERMISSIONS:
-        test = get_permission(session, name)
-        if test:
-            continue
-        try:
-            create_permission(session, name, description)
-            session.flush()
-        except IntegrityError:
-            session.rollback()
-            raise Exception("Failed to create permission: %s" % (name,))
-        session.commit()
-
-    # This group is needed to bootstrap a Grouper installation.
-    admin_group = Group.get(session, name="grouper-administrators")
-    if not admin_group:
-        admin_group = Group(
-            groupname="grouper-administrators",
-            description="Administrators of the Grouper system.",
-            canjoin="nobody",
-        )
-
-        try:
-            admin_group.add(session)
-            session.flush()
-        except IntegrityError:
-            session.rollback()
-            raise Exception("Failed to create group: grouper-administrators")
-
-        for permission_name in (GROUP_ADMIN, PERMISSION_ADMIN, USER_ADMIN):
-            permission = get_permission(session, permission_name)
+    from grouper.usecases.factory import UseCaseFactory
+
+
+class SyncDbCommand(CtlCommand):
+    """Commands to initialize the database."""
+
+    @staticmethod
+    def add_arguments(parser):
+        # type: (ArgumentParser) -> None
+        return
+
+    def __init__(self, settings, usecase_factory):
+        # type: (CtlSettings, UseCaseFactory) -> None
+        self.settings = settings
+        self.usecase_factory = usecase_factory
+
+    def run(self, args):
+        # type: (Namespace) -> None
+        usecase = self.usecase_factory.create_initialize_schema_usecase()
+        usecase.initialize_schema()
+
+        # TODO(rra): The code below will move into use cases later.
+
+        # Add some basic database structures we know we will need if they don't exist.
+        session = make_session(self.settings)
+
+        for name, description in SYSTEM_PERMISSIONS:
+            test = get_permission(session, name)
+            if test:
+                continue
+            try:
+                create_permission(session, name, description)
+                session.flush()
+            except IntegrityError:
+                session.rollback()
+                raise Exception("Failed to create permission: %s" % (name,))
+            session.commit()
+
+        # This group is needed to bootstrap a Grouper installation.
+        admin_group = Group.get(session, name="grouper-administrators")
```

Comment:
While you're here, may be worth defining "grouper-administrators" in grouper/constants.py

Reply:
Good idea. Done.